applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.cpp

```diff
@@ -31,44 +31,60 @@ KratosGeoSettlement::KratosGeoSettlement()
     }
 }
 
-int KratosGeoSettlement::RunStage(const std::string&                      rWorkingDirectory,
-                                  const std::string&                      rProjectParametersFileName,
+int KratosGeoSettlement::RunStage(const std::filesystem::path&            rWorkingDirectory,
+                                  const std::filesystem::path&            rProjectParametersFile,
                                   const std::function<void(const char*)>& ,
                                   const std::function<void(double)>&      ,
                                   const std::function<void(const char*)>& ,
                                   const std::function<bool()>&            )
 {
     KRATOS_INFO("KratosGeoSettlement") << "About to run a stage..." << std::endl;
 
-    const auto project_parameters_file_path = rWorkingDirectory + "/" + rProjectParametersFileName;
-    const auto project_parameters = InputUtilities::ProjectParametersFrom(project_parameters_file_path);
+    const auto project_parameters_file_path = rWorkingDirectory / rProjectParametersFile;
+    const auto project_parameters = InputUtilities::ProjectParametersFrom(project_parameters_file_path.generic_string());
     KRATOS_INFO("KratosGeoSettlement") << "Parsed project parameters file " << project_parameters_file_path << std::endl;
 
-    const auto model_part_name = project_parameters["solver_settings"]["model_part_name"].GetString();
-    ModelPart& model_part = mModel.CreateModelPart(model_part_name);
-    model_part.SetBufferSize(2);
-    KRATOS_INFO("KratosGeoSettlement") << "Created a model part" << std::endl;
+    if (const auto model_part_name = project_parameters["solver_settings"]["model_part_name"].GetString();
+        !mModel.HasModelPart(model_part_name)) {
+        auto& model_part = AddNewModelPart(model_part_name);
+        const auto mesh_file_name = project_parameters["solver_settings"]["model_import_settings"]["input_filename"].GetString();
+        ReadModelFromFile(rWorkingDirectory / mesh_file_name, model_part);
+    }
+
+    if (project_parameters["solver_settings"].Has("material_import_settings")) {
```

Comment:
Just to double check: could it be that the material_import_settings are initialized empty, so there is no materials_filename (or that's empty) and could that lead to issues further down the line?

Reply:
You are right: in this custom workflow we barely do any input checking. That is a fair point that we may need to investigate further. (If we feel we should improve in that area, we should also revisit `dgeoflow`, since it is similar in its set up.)

Reply:
As discussed, it's out of scope to tackle this now and the extra checks added in this PR suffice for the current purposes.